### PR TITLE
sceptre app improvements

### PR DESCRIPTION
**Pull request by GhostofGoes**
## Summary
This PR consists of changes to the sceptre app, including a new infrastructure and several Simulink-related enhancements.

## Changes
- Add `.editorconfig`. Any editors that support it, such as VSCode with an extension, will ensure correct formatting for the project.
- infrastructures.py
    - Add `BatteryInfrastructure` class
    - Remove unused method, `Infrastructure.factory()`
    - Code cleanup and formatting
- configs.py
    - Add reference to `BatteryInfrastructure`
- sceptre.py
    - Fix permissions on simulink binary (and other binaries) to ensure they're executable. This resolves a common issue where the executable permission isn't set on the source file by ensure it gets set on the inject.
    - The fields `gt` and `gt_template` are no longer required for provider metadata. This hasn't been maintained in a while according a SCEPTRE expert. They'll still be used if present, but they're no longer *required* to be there.
    - Simulink provider: Don't require `publish_endpoint` to be set for Simulink provider. If unspecified, it'll be set to a default value of `udp://*;239.0.0.1:40000`. Virtually every topology has the same value for `publish_endpoint`. This should be able to be changed if needed, but it shouldn't be required, as it's just unnecessary boilerplate. If it's configured, use the value, if not, default to the UDP multicast address and port 40000.
    - Remove unused imports
    - Fix a few missing variable references
    - Fix "description" being mis-spelled for a few injects
    - Formatting fixes and other cleanup

### CODE REVIEW

**Comment by GhostofGoes** [comment]
This is complete and ready to merge once reviewed.

**Comment by activeshadow** [comment]
I'm not familiar enough with the sceptre app to provide an adequate review.

@GhostofGoes perhaps you can request a review from someone you know is familiar with the app?

**Comment by GhostofGoes** [comment]
@mgaliar I think would be able to review it. I don't think I have the permissions required to request reviews through GitHub.

**Comment by mgaliar** [comment]
I won't be able to take a look until 10/3 at the earliest. If @c-abate has time he can probably take a look :). Otherwise, I will look at it then.

**Review by c-abate — APPROVED** [review]

**Comment by c-abate** [comment]
I've reviewed and approved, but I can't merge until @mgaliar approves as well

**Review by mgaliar — APPROVED** [review]

**Pull request merged**
